internal/publish/publish.go

```diff
@@ -65,21 +65,25 @@ type publishHandler struct {
 	authorizedAppProvider authorizedapp.Provider
 }
 
-// There is a target normalized latency for this function. This is to help prevent
-// clients from being able to distinguish from successful or errored requests.
-func (h *publishHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+type response struct {
+	status          int
+	message         string
+	metric          string
```

Comment:
It might be worth separating the metric incrementing from the response object. handleRequest() might want to increment multiple metrics during its work, and there doesn't seem to be too much advantage in deferring this to be the responsibility of the caller.

Reply:
there are still other metrics that can be incremented - but this centralizes all of the end of response metrics.